daml-model/daml/Tests/LoanConfigTest.daml

```diff
@@ -0,0 +1,300 @@
+--
+-- Copyright (c) 2022, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+--
+
+module Tests.LoanConfigTest where
+
+import DA.Assert
+import Daml.Script
+import Model.Balance
+import Model.LoanConfig
+import Tests.PartyAlloc
+
+setup = do
+    proposalTime <- getTime
+    banks@Banks{..} <- actionPartyAllocation
+    users@UserParties{..} <- userPartyAllocation
+    pure (banks, users)
+
+create_loan_should_increase_balance : Script ()
+create_loan_should_increase_balance = do
+    (Banks{..}, users@UserParties{..}) <- setup
+    submit bank11 do 
+        createCmd Balance with
+            provider = bank11
+            owner = Some users.alice
+            iban = "CH123"
+            currency = "USD"
+            amount = 0.0
+
+    loanConfigCid <- submit bank11 $ do createCmd
+        LoanConfig with
+            provider = bank11
+            accountHolder = users.alice
+            account = "CH123"
+            activeLoan = 0.0
+            balanceLoanThreshold = 1000.0
+            maxLoanThreshold = 10000.0
+            balancePaybackThreshold = 20000.0
+
+    (loanConfigCid, balanceCid) <- submit bank11 do
+        exerciseCmd loanConfigCid CreateLoan with loanAmount = None
+    
+    queryContractId bank11 loanConfigCid >>= optional 
+        (fail "LoanConfig not found") 
+        (\x -> x.activeLoan === 1000.0)
+
+    queryContractId bank11 balanceCid >>= optional 
+        (fail "Balance not found") 
+        (\x -> x.amount === 1000.0)     
+
+create_loan_should_increase_balance_partially : Script ()
+create_loan_should_increase_balance_partially = do
+    (Banks{..}, users@UserParties{..}) <- setup
+    submit bank11 do 
+        createCmd Balance with
+            provider = bank11
+            owner = Some users.alice
+            iban = "CH123"
+            currency = "USD"
+            amount = 0.0
+
+    loanConfigCid <- submit bank11 $ do createCmd
+        LoanConfig with
+            provider = bank11
+            accountHolder = users.alice
+            account = "CH123"
+            activeLoan = 0.0
+            balanceLoanThreshold = 1000.0
+            maxLoanThreshold = 10000.0
+            balancePaybackThreshold = 20000.0
+
+    (loanConfigCid, balanceCid) <- submit bank11 do
```

Comment:
A nit again: my personal taste is to only use `do` when necessary e.g. here you could just have `$` instead. But not important.